README.md

```diff
@@ -1,16 +1,159 @@
+the _assignment operator_; and then an expression or a constant, which we call
+`rhs` for "right-hand side."
+
+Here's an example:
+
+```ruby
+height_in_centimeters = 180
+```
+
+The left-hand side is a name. Often, programmers call it the "variable name."
+The "LHS" **can never** be an expression. It is called a "bare word." Bare
+words (also: "barewords") are not data like `180`, nor are they expressions
+like `1 + 1`, nor are they words that are special to Ruby. They're words that
+***we*** introduce. For Ruby, `height_in_centimeters` is a bare word. For a
+human baby, `Ma-Ma` is a bare word.
+
+These bare-word names are made to "point" to the RHS. This operator is known as
+the _assignment operator_. The _assignment operator_ should not be confused
+with `=` which we know from daily life and arithmetic which means "is equal
+to."
+
+> **ASIDE**: Some programming language authors find "`=` means assignment" to
+> be very confusing.  The Go programming language from Google uses `:=` to mean
+> assignment as does the Pascal programming language.
+
+The RHS should be an _expression_, even if it's our friend the lowly _constant
+expression_. To associate this to learning to speak:
+
+```text
+Ma-ma = <this person in front of me>
+```
+
+That's the basics of the assignment expression!
+
+## Define Mutability / Immutability
+
+A variable is said to be mutable. We can apply multiple assignment expressions
+to it.
+
+Many years ago I was:
+
+```ruby
+height_in_centimeters = 50
+```
+
+But today I am:
+
+```ruby
+height_in_centimeters = 180
+```
+
+We can try these out in IRB:
+
+(animation)
+
+Sometimes, we might want to make a variable name permanent, it can only be
+assigned ONCE. We do this by writing a "symbolic constant." In Ruby, symbolic
+constants are written in all caps:
+
+```ruby
+SPEED_OF_LIGHT = 180000
+```
+
+If, after this you try to set `SPEED_OF_LIGHT` to something else, Ruby will
+complain.
+
+```ruby
+SPEED_OF_LIGHT = 1800000
+warning: already initialized constant SPEED_OF_LIGHT
+```
+
+Actually, our "baby" example is probably:
+
+```ruby
+MA_MA = <the lady that gave me life who is gesturing in front of me>
+```
+
+## Constant versus "Symbolic Constants" and "Values"
+
+We need to clear up some vocabulary.
+
+We've been calling `23` and `255` _constants_, because that's what the
+definition of _expression_ called them. And this is absolutely true and
+correct.
+
+**But** programmers have come to say, "constant" is short for "symbolic
+constant" &mdash; like we just learned. What _we_ have been calling _constants_
+many programmers call "values" or "scalar values."
+
+## Return Value of an _Assignment Expression_
+
+It's interesting, but the return value of an _assignment expression_ is the
+evaluated result of the RHS.
+
+Let's do a table again:
+
+|Expession|Action|Notes|
+|---------|------|-----|
+|`x = 2 * (y = 3 * 2)`| Evaluate innermost expression in `()`| PEMDAS rules|
```

Comment:
this might seem a little strange with the double assignment. Would it still make sense if it was separated into `y=3` and `x = 2 * ( y * 2)`? or `y=3*2` and `x= 2 * y` or something similar

Reply:
I'm trying to get that the return value of an assignment expression can be integrated into another expression.

Reply:
Is there some other improvement to do, given this? Or is it too advanced?